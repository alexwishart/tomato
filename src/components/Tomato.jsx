import React from 'react';
import styled from 'styled-components';
import MainInfo from './MainInfo';

const Tomato = () => {
  return (
    <Container>
      <Title>
        <h1>The Tomato</h1>
      </Title>
      <ContentContainer>
        <MainInfo />
      </ContentContainer>
    </Container>
  );
};

const Container = styled.div`
  padding: 0 2rem;

  font-family: Arial, Helvetica, sans-serif;
  text-align: left;
`;

const Title = styled.div`
  width: 100%;
`;

const ContentContainer = styled.div`
  display: flex;
`;

export default Tomato;
