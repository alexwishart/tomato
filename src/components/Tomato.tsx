import React from 'react';
import styled from 'styled-components';
import MainInfo from './MainInfo';
import NutritionalInfo from './NutritionalInfo';
import TabView from './TabView';
import { mainContent } from '../content/main';
import { tomatoTabs } from '../content/tabInfo';

const Tomato = () => {
  return (
    <Container>
      <Title>
        <h1>{mainContent.title}</h1>
      </Title>
      <ContentContainer>
        <MainInfo />
        <NutritionalInfo />
      </ContentContainer>
      <TabView content={tomatoTabs} />
    </Container>
  );
};

const Container = styled.div`
  padding: 0 2rem;

  font-family: Arial, Helvetica, sans-serif;
  text-align: left;
`;

const Title = styled.div`
  width: 100%;
`;

const ContentContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export default Tomato;
