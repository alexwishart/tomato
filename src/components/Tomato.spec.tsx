import React from 'react';
import { render } from '@testing-library/react';
import Tomato from './Tomato';

jest.mock('./TabView', () => {
  return jest.fn(() => <div id="TabView" />);
});

describe('The Tomato component', () => {
  it('should render correctly', () => {
    expect(render(<Tomato />)).toMatchSnapshot();
  });
});
