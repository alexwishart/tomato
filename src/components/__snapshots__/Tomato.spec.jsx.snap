// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tomato component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSgupP bhWUnM"
      >
        <div
          class="sc-gKsewC jwAxTp"
        >
          <h1>
            The Tomato
          </h1>
        </div>
        <div
          class="sc-iBPRYJ eBvofg"
        >
          <div
            class="sc-bdfBwQ cYPkhV"
          >
            <img
              alt="Image of tomato"
              class="sc-gsTCUz fBVUqk"
              src="tomato.jpg"
            />
            <h2>
              Varieties
            </h2>
            <p>
              There are around 7500 tomato varieties grown for various purposes. Heirloom tomatoes are becoming increasingly popular, particularly among home gardeners and organic producers, since they tend to produce more interesting and flavorful crops at the cost of disease resistance and productivity. In 1973, Israeli scientists developed the world's first long shelf-life commercial tomato varieties.
            </p>
            <p>
              Tomato varieties are roughly divided into several categories, based mostly on shape and size.
            </p>
            <ul>
              <li>
                "Slicing"/"globe" tomatoes
              </li>
              <li>
                Beefsteak tomatoes
              </li>
              <li>
                Oxheart tomatoes
              </li>
              <li>
                Plum tomatoes
              </li>
              <li>
                Pear tomatoes
              </li>
              <li>
                Cherry tomatoes
              </li>
              <li>
                Grape tomatoes
              </li>
              <li>
                Campari tomatoes
              </li>
            </ul>
          </div>
          <div
            class="sc-dlfnbm gFdIRu"
          >
            <h3>
              Red tomatoes, raw
            </h3>
            <p>
              Nutritional value per 100 g (3.5 oz)
            </p>
            <table>
              <tbody>
                <tr>
                  <th
                    class="sc-hKgILt jtmEwi"
                  >
                    Energy
                  </th>
                  <th
                    class="sc-hKgILt jtmEwi"
                  >
                    74kJ (18 kcal)
                  </th>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Carbohydrates
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    3.9 g
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    - Sugars
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    2.6 g
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    - Dietary fiber
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    1.2 g
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Fat
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    0.2 g
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Protein
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    0.9 g
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Water
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    94.5 g
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Vitamin A equiv.
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    42 µg (5%)
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    - lutein and zeaxanthin
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    123 µg
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Vitamin C
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    14 mg (17%)
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Vitamin E
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    0.54 mg (4%)
                  </td>
                </tr>
                <tr>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    Potassium
                  </td>
                  <td
                    class="sc-eCssSg PDepf"
                  >
                    237 mg (5%)
                  </td>
                </tr>
              </tbody>
            </table>
            <p>
              Percentages are relative to US recommendations for adults
            </p>
            <p>
              Source:
               
              <a
                href="http://www.nal.usda.gov/fnic/foodcomp/search/"
              >
                 
                USDA Nutrient Database
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSgupP bhWUnM"
    >
      <div
        class="sc-gKsewC jwAxTp"
      >
        <h1>
          The Tomato
        </h1>
      </div>
      <div
        class="sc-iBPRYJ eBvofg"
      >
        <div
          class="sc-bdfBwQ cYPkhV"
        >
          <img
            alt="Image of tomato"
            class="sc-gsTCUz fBVUqk"
            src="tomato.jpg"
          />
          <h2>
            Varieties
          </h2>
          <p>
            There are around 7500 tomato varieties grown for various purposes. Heirloom tomatoes are becoming increasingly popular, particularly among home gardeners and organic producers, since they tend to produce more interesting and flavorful crops at the cost of disease resistance and productivity. In 1973, Israeli scientists developed the world's first long shelf-life commercial tomato varieties.
          </p>
          <p>
            Tomato varieties are roughly divided into several categories, based mostly on shape and size.
          </p>
          <ul>
            <li>
              "Slicing"/"globe" tomatoes
            </li>
            <li>
              Beefsteak tomatoes
            </li>
            <li>
              Oxheart tomatoes
            </li>
            <li>
              Plum tomatoes
            </li>
            <li>
              Pear tomatoes
            </li>
            <li>
              Cherry tomatoes
            </li>
            <li>
              Grape tomatoes
            </li>
            <li>
              Campari tomatoes
            </li>
          </ul>
        </div>
        <div
          class="sc-dlfnbm gFdIRu"
        >
          <h3>
            Red tomatoes, raw
          </h3>
          <p>
            Nutritional value per 100 g (3.5 oz)
          </p>
          <table>
            <tbody>
              <tr>
                <th
                  class="sc-hKgILt jtmEwi"
                >
                  Energy
                </th>
                <th
                  class="sc-hKgILt jtmEwi"
                >
                  74kJ (18 kcal)
                </th>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Carbohydrates
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  3.9 g
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  - Sugars
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  2.6 g
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  - Dietary fiber
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  1.2 g
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Fat
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  0.2 g
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Protein
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  0.9 g
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Water
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  94.5 g
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Vitamin A equiv.
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  42 µg (5%)
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  - lutein and zeaxanthin
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  123 µg
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Vitamin C
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  14 mg (17%)
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Vitamin E
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  0.54 mg (4%)
                </td>
              </tr>
              <tr>
                <td
                  class="sc-eCssSg PDepf"
                >
                  Potassium
                </td>
                <td
                  class="sc-eCssSg PDepf"
                >
                  237 mg (5%)
                </td>
              </tr>
            </tbody>
          </table>
          <p>
            Percentages are relative to US recommendations for adults
          </p>
          <p>
            Source:
             
            <a
              href="http://www.nal.usda.gov/fnic/foodcomp/search/"
            >
               
              USDA Nutrient Database
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
